app/controllers/conversations_controller.rb

```diff
@@ -1,8 +1,23 @@
 class ConversationsController < ApplicationController
   before_action :signed_in_flash, only: [:create]
   # before_action :participant?, only: [:show]
+
   def index
-    @conversations = Conversation.all.order("created_at DESC")
+    if params[:order] == "totalpoints"
+      @conversations = Kaminari.paginate_array(Conversation.all.sort{ |b,a| a.sender.average_score + a.recipient.average_score <=> b.sender.average_score + b.recipient.average_score }).page(params[:page]).per(10)
+    elsif params[:order] == "message"
+        @conversations = Kaminari.paginate_array(Conversation.all.sort{ |b,a| a.messages.count <=> b.messages.count }).page(params[:page]).per(10)
+    elsif params[:order] == "points"
+        @conversations = Kaminari.paginate_array(Conversation.all.sort{ |b,a| a.sender.total + a.recipient.total <=> b.sender.total + b.recipient.total }).page(params[:page]).per(10)
+    elsif params[:order] == "sender"
+      @conversations = Kaminari.paginate_array(Conversation.all.sort{ |b,a| a.sender.average_score || a.recipient.average_score <=> b.sender.average_score || b.recipient.average_score }).page(params[:page]).per(10)
```

Comment:
Line is too long. [214/80]<br>Space missing after comma.<br>Space missing to the left of {.